internal/backoff/backoff.go

```diff
@@ -60,7 +58,32 @@ func WaitUntil(
 	eb.RandomizationFactor = backoffSettings.RandomizationFactor
 	eb.Multiplier = backoffSettings.Multiplier
 
-	backoffWithContext := backoff.WithContext(eb, ctx)
+	backoffWithContext := Context(ctx, backoffSettings)
 
 	return backoff.Retry(operation, backoffWithContext)
 }
+
+// WaitUntilWithData Implementation of backoff operations that increases the back off period for each retry
+// attempt using a randomization function that grows exponentially. This does not allow for parameters.
+// nolint: ireturn
+func WaitUntilWithData[T any](
```

Comment:
Is this function is being used anywhere?

Reply:
not yet but I added it in because every time I go to look at WaitUntil I think I have to write a function like this